packages/legacy-client/src/client/convertWireToOsdkObject.ts

```diff
@@ -87,29 +91,30 @@ function createPrototype<
  */
 const cache = new Map<string, Map<string, any>>();
 export function convertWireToOsdkObject<
-  T extends keyof O["objects"] & string,
+  T extends ObjectTypesFrom<O> & string,
   O extends OntologyDefinition<any>,
 >(
-  context: ClientContext,
-  obj: OsdkLegacyPropertiesFrom<O, T> & OntologyObject<T>,
+  client: ThinClient<O>,
+  apiName: T,
+  obj: OntologyObjectV2,
```

Comment:
This change right here is pretty important. It effectively gives us `Record<string, any>`, and we're depending on it to contain `__rid` and `__primaryKey` on it.

Reply:
Yeah unfortunately we need to make that assumption at some point